File: leetcode/1480.cpp
// Runtime: 3 ms, faster than 76.26% of C++ online submissions for Running Sum of 1d Array.
// Memory Usage: 8.6 MB, less than 38.38% of C++ online submissions for Running Sum of 1d Array.
class Solution {
 public:
  vector<int> runningSum(vector<int>& nums) {
    vector<int> res;
    res.push_back(0);
    for (auto n : nums) {
      res.push_back(n + *(res.end() - 1));
    }
    res.erase(res.begin());
    return res;
  }
};

File: leetcode/1480.py
"""
Runtime: 40 ms, faster than 60.30% of Python3 online submissions for Running Sum of 1d Array.
Memory Usage: 14.3 MB, less than 87.98% of Python3 online submissions for Running Sum of 1d Array.
"""


class Solution:
    def runningSum(self, nums: List[int]) -> List[int]:
        for i in range(len(nums) - 1):
            nums[i + 1] = nums[i + 1] + nums[i]

        return nums
